RHDEVDOCS 5719 running Buildah as nonroot with user namespaces (#79325)
openshift/openshift-docs: secure/unprivileged-building-of-container-images-using-buildah.adoc

toc::[]

Running {pipelines-shortname} as the root user on a container can expose the container processes and the host to other potentially malicious resources. You can reduce this type of exposure by running the workload as a specific non-root user in the container.

> In most cases, creating a custom task for building the image and configuring user namespaces in this task enables running Buildah without root privileges.

Reviewer: just a suggestion, i think it reads a little easier this way:
```suggestion
In most cases, you can run Buildah without root privileges by creating a custom task for building the image and configuring user namespaces in this task.
```

> If your image does not build successfully using this configuration, you can define custom service account (SA) and security context constraint (SCC); however, in this option, you must enable the Buildah step to raise its privileges (`allowPrivilegeEscalation: true`).

Reviewer: i'm not sure if the options you're referring to is the latter SCC or both of them. my suggestion is to make that explicit (i guessed it's for both of them, but i might be wrong 😄)
```suggestion
If your image does not build successfully using this configuration, you can define a custom service account (SA) and a security context constraint (SCC). However, in either of these cases, you must enable the Buildah step to raise its privileges (`allowPrivilegeEscalation: true`).
```

Author: This is a single option. I did some slight rewording to make this fact clearer.

include::modules/op-nonroot-buildah-user-namespaces.adoc[leveloffset=+1]

[id="buildah-nonroot-sa-scc"]
== Running Buildah as a non-root user by defining a custom SA and SCC

To run builds of container images using Buildah as a non-root user, you can perform the following steps:

> * Define custom service account (SA) and security context constraint (SCC).
> * Configure Buildah to use the `build` user with id `1000`.
> * Start a task run with a custom config map, or integrate it with a pipeline run.

Reviewer: outside the scope of your changes and this PR, but this list should probably be ordered if they are to be followed sequentially:
```suggestion
. Define a custom service account (SA) and security context constraint (SCC).
. Configure Buildah to use the `build` user with id `1000`.
. Start a task run with a custom config map, or integrate it with a pipeline run.
```

Author: Outside the scope indeed - I would not want to touch this part right now as any reworking is likely to require a separate review cycle.
